indico/MaKaC/PDFinterface/conference.py

```diff
@@ -417,65 +415,25 @@ class ContributionBook(PDFLaTeXBase):
     _tpl_filename = "contribution_list_boa.tpl"
 
     def _sort_contribs(self, contribs, sort_by, aw):
-        if sort_by == "boardNo":
-            try:
-                return sorted(contribs, key=lambda x: int(x.getBoardNumber()))
-            except ValueError, e:
-                raise MaKaCError(
-                    _("In order to generate this PDF, all the contributions must contain a board number "
-                      "and it must only contain digits. There is a least one contribution with a wrong board number."))
-
-        elif sort_by == "schedule":
-            # TODO: this ignores contribution list param
-            # Maybe it would be worth ordering the incoming contribs
-            # by schedule time, if such a use case makes sense
-
-            res = []
-            for entry in self._conf.getSchedule().getEntries():
-                entry_owner = entry.getOwner()
-                if isinstance(entry, schedule.LinkedTimeSchEntry) and isinstance(entry_owner, conference.SessionSlot):
-                    if entry_owner.canAccess(aw):
-                        for slotentry in entry_owner.getSchedule().getEntries():
-                            owner = slotentry.getOwner()
-                            if isinstance(owner, conference.Contribution):
-                                res.append(owner)
-
-                elif isinstance(entry, schedule.LinkedTimeSchEntry) and isinstance(entry_owner, conference.Contribution):
-                    res.append(entry_owner)
-            return res
-        else:
-            fc = FilterCriteria(self._conf, {
-                "status": [ContribStatusList.getId(conference.ContribStatusSch),
-                           ContribStatusList.getId(conference.ContribStatusNotSch)]
-                })
-            sc = contribFilters.SortingCriteria((sort_by,))
-            f = filters.SimpleFilter(fc, sc)
-
-            res = []
-            for contrib in f.apply(contribs):
-                res.append(contrib)
-            return res
+        attr_getter = attrgetter('title')
```

Comment:
```
attr = {'boardNo': 'board_number', 'schedule': 'start_dt'}.get(sorty_by, 'title')
return sorted(contribs, key=attrgetter(attr))
```

(if there's a `sort_by` value for title I'd use that and `[]` instead of `get()`)